xds/internal/client/xds.go

```diff
@@ -571,40 +571,35 @@ func unmarshalClusterResource(r *anypb.Any, logger *grpclog.PrefixLogger) (strin
 	return cluster.GetName(), cu, nil
 }
 
-func clusterTypeFromCluster(cluster *v3clusterpb.Cluster) (ClusterType, string, []string, error) {
+func clusterTypeFromCluster(cluster *v3clusterpb.Cluster) (ClusterType, string, string, []string, error) {
```

Comment:
This function seems to be doing a bit too much. It is not only determining the cluster type, but also getting the cluster name, eds service name, list of clusters (for the aggregate case).

Can we simplify things here? Some thoughts that come to mind are:
- Make this function only return the cluster type, and have a switch case in the caller based on the cluster type and set the appropriate fields in the `ClusterUpdate` directly
- Pass the `ClusterUpdate` as a pointer to this function and have it directly set things up. Rename the function appropriately if we are doing that.

Reply:
I moved code around and deleted this function.
PTAL.